src/main/java/com/example/petbuddybackend/entity/user/Client.java

```diff
@@ -44,6 +45,10 @@ public class Client {
     @Builder.Default
     private Set<ClientNotification> notifications = new HashSet<>();
 
+    @OneToMany(mappedBy = "client", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    @Builder.Default
+    private List<Care> cares = new ArrayList<>();
```

Comment:
Nie wiem czy bym dawał tutaj `CascadeType.ALL`

Reply:
wiem, tez bym nie dawal, ale opieka tez musi miec przypisanego klienta, a skoro usuwamy klienta to jest niespojnosc.
Poza tym i tak zakladamy ze klient ani opiekun nie bedzie usuwany, a skoro tak to opieki tez zostaną

Reply:
Chyba ze samego remove dac

Reply:
Właśnie czytałem coś o tym usuwaniu. Jest coś takiego jak `soft delete` i myślę że by to było dobre rozwiązanie dla nas. 

Polegałoby to na tym, że:
- dezaktywujemy konto użytkownika, czyli nie będzie mógł on się uwierzytelniać i nie będzie można zrobić GETa na jego dane itp.
- można usunąć encje niepotrzebne np. recenzje a historyczne zostawić np. opieki przez co potencjalny klient będzie miał dostęp do opiek i zobaczy z kim je miał
- Dane wrażliwe dla usuniętego użytkownika można usunąć np. adres zostawić pusty

Bardziej to piszę jako ciekawostkę bo prawdopodobnie braknie czasu na implementację tego, ale jeśli wydaje ci się ok to podejście to można by było usunąć cascade delete z opiek dzięki temu i dać furtkę na implementację usuwania konta

Reply:
tylko co by opieka miala przypisane wtedy w miejsce opieki?
Plus mozna to zrobic dodajac kolumne isActive do AppUser i wtedy dezaktywujemy bez usuwania

Reply:
Ale w takim razie to też nie byłoby usuwanie samej encji tylko usunięcie danych z niektórych kolumn, więc i tak cascade się nie wywola

Reply:
Ale mogę te kaskady tak czy inaczej usunac

Reply:
No w sumie tak xd